core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.cpp

```diff
@@ -100,11 +103,23 @@ void OpenMPTargetExec::resize_scratch(int64_t team_size, int64_t shmem_size_L0,
   const int64_t shmem_size =
       shmem_size_L0 + shmem_size_L1;  // L0 + L1 scratch memory per team.
   const int64_t padding = shmem_size * 10 / 100;  // Padding per team.
+
+  // Maximum active teams possible.
+  // The number should not exceed the maximum in-flight teams possible or the
+  // league_size.
+  int max_active_teams =
+      std::min(OpenMPTargetExec::MAX_ACTIVE_THREADS / team_size, league_size);
+
+  // max_active_teams is the number of active teams on the given hardware.
+  // We set the number of teams to be twice the number of max_active_teams for
+  // the compiler to pick the right number in its case.
+  omp_set_num_teams(max_active_teams * 2);
```

Comment:
`OpenMPTargetExec::resize_scratch` seems like a weird place to set this.

Reply:
`resize_scratch` is called for every instance of hierarchical parallelism since we want to control the number of teams generated. And since we allocate data for max_active_teams in the `resize_scratch` routine, I set the number of teams there so that we don't need to save it somewhere or recalculate it.